### DISCUSSION TITLE
'Expand' action using API doesn't register as a "click"

### DISCUSSION THREAD

**Discussion by d-danger** [YOURLS/YOURLS#3784] [General]
Using the API to expand a keyword to its original URL doesn't register itself as a click.  I realise this is probably intended behaviour, but is there a way to register it as a click in the yourls-api.php file when 'expand' action is called?

**Comment by dgw** [comment]
> I realise this is probably intended behaviour

Indeed.

> is there a way to register it as a click in the yourls-api.php file when 'expand' action is called?

Not there, no. But you can hook into the `api_expand` filter here:

https://github.com/YOURLS/YOURLS/blob/41ff6adf5404e762fc1700e3e7091595b6069501/includes/functions-api.php#L240

A plugin can do whatever it wants with the result—including calling [`yourls_update_clicks()`](https://github.com/YOURLS/YOURLS/blob/41ff6adf5404e762fc1700e3e7091595b6069501/includes/functions.php#L90) with the `$return['keyword']` being expanded—as long as it returns the original action inputs at the end (this is a "filter", not an "action", so it must return the original value even if not modified).
